app/main/services/window-process-management-service.ts

```diff
@@ -162,6 +162,21 @@ export class WindowProcessManagementService extends Eventable<IWindowProcessMana
       });
     }
 
+     // Adding custom css
+     const insertedKeys = this._preferenceService.get("_insertedCustomCssKey") as { windowId: string, key: string }[]
```

Comment:
Why do we need to store the key in preference?

If I understand correctly,

1. we store the customCSS in preference.
2. When we create the window, we use `insertCSS` to insert the CSS and get the `key`.
3. We store it into a private member of the windowManagemetService, say `private _insertedKeys = []`

It seems that we don't need to store it in the preference.

Reply:
Although I admit that I didn't think much about it when storing this key in the preferences, I've come up with a fairly good excuse: 

Let's assume that in the future, we want to insert custom CSS from multiple different sources. For instance, we might need to provide multiple independent text boxes to insert corresponding CSS for different pages (I know this can easily be done by defining different selectors in CSS, but let's assume that the scale of custom CSS in the future might be so large that decoupling is necessary). This means we would need to call `insertCss` multiple times, generating different keys. At this point, the component in the preferences responsible for managing the inserted CSS would need to know how many and which CSS fragments have already been inserted. If the inserted keys are stored in the window management service, it would be difficult for the preferences to know which CSS has already been inserted.

This is just an excuse for now. We could certainly store these keys in the window management service. You can think about it, and I'll go with whatever decision you make.

Reply:
I mean, yes we need to store the key-css pair. But I think it's unnecessary to persist them in the preference. Because the CSS will always be inserted when we create a window.

Suppose we support inserting CSS for multiple pages, even multiple windows, we should persist something like this in the preference:
```
{
  windowIdA: {
    cssTextField1: "CSS 1",
    cssTextField2: "CSS 2"
  },
  windowIdB: {
    cssTextField1: "CSS 1",
    cssTextField2: "CSS 2"
  },
}
```

When we open paperlib, we get the CSSs of all windows, and then insert them. For example, for the windowIdA, we should have 2 keys. now we store these two keys with their corresponding window id and text field id in the private member `_insertedCustomCssKey`.

When we change the CSS of `windowIdA-cssTextField1`, we do the `removeInsertedCSS` first as we know the window id and the textfield id. Then do the `insertCSS`

But in your implementation, I think everytime we change the CSS, we need to restart Paperlib to take effect right?
So I think we don't need the `removeInsertedCSS` part as we only create the window once.

Reply:
Are we going to store the window specific CSS in a structured field like: 
```javascript
customCSS: {
  windowIdA: {
    cssTextField1: "CSS 1",
    cssTextField2: "CSS 2"
  },
  windowIdB: {
    cssTextField1: "CSS 1",
    cssTextField2: "CSS 2"
  },
}
```
If so, the current logic that proxies the preference state cannot deep copy the inner property of the `customCSS` object. This means that in the UI, when VUE tries to use a model like `v-model="prefState.customCSS.windowIdA.cssTextField1"`, it will get an `undefined` from the `prefState`. Changing the `prefState` proxy logic will introduce breaking changes to the code. Another solution is to store the CSS of the different windows in the first level of the `preferenceState`.

Let me know about your opinion about this :)

Reply:
Hi,

basically, we only have two windows --- the main window, and the reference quickpaste window. And we have no plan to add new windows in the near future. 

We can only provide two `textarea` in the UI. One is for the main window, the other one is for the quickpaste window.

We only store two new strings in the config.json such as : `customCSSForMain` and `customCSSForQuickpaste`.

That's enough for now.